### DISCUSSION TITLE
Reason for torch.set_num_threads(1) ?

### DISCUSSION THREAD

**Discussion by TechInterMezzo** [snakers4/silero-vad#484] [Q&A] [answered]
Why is torch.set_num_threads(1) used in the examples?
Does it improve performance for the model or does it prevent bugs?

**Comment by snakers4** [comment, marked as answer]
The VAD does not require more compute than 1 CPU thread by design.
Using more threads (or by default - all CPU threads) may cause adverse effects.

**Reply by TechInterMezzo** [reply]
Does that have any (negative) effect when using the GPU?

**Reply by snakers4** [reply]
These are CPU threads, I believe not, but the VAD is not meant to run on GPU

**Reply by TechInterMezzo** [reply]
Okay. I read this in the README:

> One audio chunk (30+ ms) takes less than 1ms to be processed on a single CPU thread. Using batching or **GPU** can also improve performance considerably. Under certain conditions ONNX may even run up to 4-5x faster.

So I thought it would be even better on GPU with batching.
